Integrations/src/main/java/io/deephaven/integrations/learn/Gatherer.java

```diff
@@ -0,0 +1,543 @@
+package io.deephaven.integrations.learn;
+
+import io.deephaven.chunk.*;
+import io.deephaven.chunk.attributes.Values;
+import io.deephaven.engine.rowset.RowSequence;
+import io.deephaven.engine.table.ChunkSource;
+import io.deephaven.engine.table.ColumnSource;
+import io.deephaven.engine.table.SharedContext;
+import io.deephaven.util.SafeCloseableList;
+
+/**
+ * Gatherer takes Deephaven columnar data and places it into a buffer to be used by Python. The Python object will take
+ * data from the buffer and use it to construct a 2d array of specified size.
+ */
+public class Gatherer {
+
+    private static final int COPY_CHUNK_SIZE = 2048;
+
+    /**
+     * Copy data from a table into a 2d tensor of Booleans in column-major order.
+     *
+     * @param rowSeq indices of the rows of the table to put into the tensor
+     * @param columnSources columns of data to put into the tensor
+     * @return contiguous RAM allocated for the tensor. When a numpy tensor is passed in for this argument, jpy will
+     *         handle passing the memory reference as a 1d java array here.
```

Comment:
```suggestion
     * @return contiguous RAM allocated for the tensor.
```
No way to pass in an array.  Apply this to all other locations.